src/Symfony/Component/HttpClient/CachingHttpClient.php

```diff
@@ -71,7 +67,7 @@ public function request(string $method, string $url, array $options = []): Respo
         $url = implode('', $url);
         $options['extra']['no_cache'] = $options['extra']['no_cache'] ?? !$options['buffer'];
 
-        if ($options['extra']['no_cache'] || !empty($options['body']) || !\in_array($method, ['GET', 'HEAD', 'OPTIONS'])) {
+        if (!empty($options['body']) || $options['extra']['no_cache'] || !\in_array($method, ['GET', 'HEAD', 'OPTIONS'])) {
```

Comment:
I dont really understand this change. Why is the new version better?

Reply:
Same reason as in https://github.com/symfony/symfony/pull/31018#discussion_r273341842